include/swift/SIL/OwnershipUtils.h

```diff
@@ -264,10 +267,14 @@ struct BorrowingOperand {
 
   BorrowingOperand(Operand *op)
       : op(op), kind(BorrowingOperandKind::get(op->getUser()->getKind())) {
-    if (kind == BorrowingOperandKind::Branch
-        && op->get().getOwnershipKind() != OwnershipKind::Guaranteed) {
+    auto ownership = op->getOperandOwnership();
+    if (ownership != OperandOwnership::Borrow
+        && ownership != OperandOwnership::Reborrow) {
+      // consuming applies and branch arguments are not borrowing operands.
       kind = BorrowingOperandKind::Invalid;
+      return;
     }
+    assert(kind != BorrowingOperandKind::Invalid && "missing case");
```

Comment:
With this can we now eliminate other kinds in BorrowingOperandKind: BeginApply, TryApply etc ?

Reply:
@meg-gupta A guaranteed apply argument is still considered a "Borrow" here. I think BorrowingOperand is useful for BeginApply because that's a scoped instruction, so it's important to call `visitScopeEndingUses` in that case.

I think it is a problem that BorrowingOperand is valid for Apply/TryApply/Yield, but  `visitScopeEndingUses` does nothing there. That will lead to bugs where we miss the use completely. I'll see if it's easy to fix that in this PR too.

Reply:
Well, it looks like we're always careful before calling `visitScopeEndUses` to also visit the instruction that begins the scope too. So the behavior doesn't need to be changed, but I can add a comment in the next PR.

Reply:
SGTM